pkg/apis/application/v1alpha1/types.go

```diff
@@ -1174,16 +1174,16 @@ type SyncPolicyAutomated struct {
 	AllowEmpty bool `json:"allowEmpty,omitempty" protobuf:"bytes,3,opt,name=allowEmpty"`
 }
 
-// SyncStrategy controls the manner in which a sync is performed
-type SyncStrategy struct {
+// SyncOperationStrategy controls the manner in which a sync is performed
+type SyncOperationStrategy struct {
```

Comment:
I had to rename this because it was colliding the SyncStrategy CRD. I considered using `ApplicationSetSyncStrategy` for the CRD and leaving this one as `SyncStrategy` but I think it is better to have `SyncStrategy` as the front-facing CRD and have the longer name for this internal type. Open to suggestions